pkg/pod/network_test.go

```diff
@@ -0,0 +1,453 @@
+		},
+		{
+			annotationName: "test",
+			macAddress:     "",
+			ipAddr:         []string{"192.168.1.1/24"},
+			expectedNil:    true,
+		},
+		{
+			annotationName: "test",
+			macAddress:     "00-B0-D0-63-C2-26",
+			ipAddr:         []string{},
+			expectedNil:    true,
+		},
+	}
+	for _, testCase := range testCases {
+		annotation := StaticIPAnnotationWithMacAddress(testCase.annotationName, testCase.ipAddr, testCase.macAddress)
+		if testCase.expectedNil {
+			assert.Nil(t, annotation)
+		} else {
+			assert.NotNil(t, annotation)
+		}
+	}
+}
+
+func TestPodNetworkStaticIPAnnotationWithMacAndNamespace(t *testing.T) {
+	testCases := []struct {
+		annotationName string
+		macAddress     string
+		namespace      string
+		expectedNil    bool
+	}{
+		{
+			annotationName: "test",
+			macAddress:     "00-B0-D0-63-C2-26",
+			namespace:      "test",
+			expectedNil:    false,
+		},
+		{
+			annotationName: "",
+			macAddress:     "00-B0-D0-63-C2-26",
+			namespace:      "test",
+			expectedNil:    true,
+		},
+		{
+			annotationName: "test",
+			macAddress:     "",
+			namespace:      "test",
+			expectedNil:    true,
+		},
+		{
+			annotationName: "test",
+			macAddress:     "00-B0-D0-63-C2-26",
+			namespace:      "",
+			expectedNil:    true,
+		},
+	}
+	for _, testCase := range testCases {
+		annotation := StaticIPAnnotationWithMacAndNamespace(testCase.annotationName, testCase.namespace, testCase.macAddress)
+		if testCase.expectedNil {
+			assert.Nil(t, annotation)
+		} else {
+			assert.NotNil(t, annotation)
+		}
+	}
+}
+
+func TestPodNetworkStaticIPAnnotationWithInterfaceMacAndNamespace(t *testing.T) {
+	testCases := []struct {
+		annotationName string
+		namespace      string
+		intName        string
+		macAddress     string
+		expectedNil    bool
+	}{
+		{
+			annotationName: "test",
+			namespace:      "test",
+			intName:        "eth0",
+			macAddress:     "00-B0-D0-63-C2-26",
+			expectedNil:    false,
+		},
+		{
+			annotationName: "",
+			namespace:      "test",
+			intName:        "eth0",
+			macAddress:     "00-B0-D0-63-C2-26",
+			expectedNil:    true,
+		},
+		{
+			annotationName: "test",
+			namespace:      "",
+			intName:        "eth0",
+			macAddress:     "00-B0-D0-63-C2-26",
+			expectedNil:    true,
+		},
+		{
+			annotationName: "test",
+			namespace:      "test",
+			intName:        "",
+			macAddress:     "00-B0-D0-63-C2-26",
+			expectedNil:    true,
+		},
+		{
+			annotationName: "test",
+			namespace:      "test",
+			intName:        "eth0",
+			macAddress:     "",
+			expectedNil:    true,
+		},
+	}
+	for _, testCase := range testCases {
+		annotation := StaticIPAnnotationWithInterfaceMacAndNamespace(
+			testCase.annotationName, testCase.namespace, testCase.intName, testCase.macAddress)
+		if testCase.expectedNil {
+			assert.Nil(t, annotation)
+		} else {
+			assert.NotNil(t, annotation)
+		}
+	}
+}
+
+func TestPodNetworkStaticIPBondAnnotationWithInterface(t *testing.T) {
+	testCases := []struct {
```

Comment:
There's no test for when ipAddrBond is empty. There is a test for when it is invalid but not empty